framework/src/predictors/AdamsPredictor.C

```diff
@@ -73,6 +73,10 @@ AdamsPredictor::apply(NumericVector<Number> & sln)
   // that gets called on time step begin.
   // That means that history control must go here.
   historyControl();
+
+  if (!shouldApply())
```

Comment:
Maybe split this up... give it an `initialize()`, `execute()` (or `apply()`) and `finalize()` just like other objects in MOOSE?

`initialize()` could be called all the time... then `shouldApply()` will be checked... the last two only get called if `shouldApply()` returns true.

Just a suggestion.  If no one else thinks it matters then it's probably fine how it is ;-)